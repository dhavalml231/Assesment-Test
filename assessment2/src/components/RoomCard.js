import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import classNames from 'classnames';
import { updateRoomConfig, setNoOfRooms } from '../actions'

const RoomCardRoot = styled.div`
    margin: 10px;
    width: 180px;
    height: 150px;
    border: 3px solid #e0e0e0;
    border-radius: 10px;
    overflow:hidden;
    div.title{
        background-color: #e0e0e0;
        line-height: 30px;
        height: 30px;
        padding-left: 5px;
        font-weight: 600;
        font-size: 14px;
        display:grid;
        grid-template-columns:20px auto;
    }
    &.disabled{
        border: 3px solid #9fa8da;
        background-color: #c5cae9;
        div.title{
            background-color:inherit;
        }
    }
`;
const Content = styled.div`
    height: 110px;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 11px;
    align-items: center;
    width: 140px;
    margin: 0 auto;
    font-size: 15px;
    font-weight: 100;
`;
export const RoomCard = ({ roomNo, noOfRooms, occupancy = {}, dispatch }) => {

    const disabled = roomNo > noOfRooms;
    const cardClassName = classNames({ 'disabled': disabled });
    const checkBoxId = `${roomNo}_chkBox`;

    const checked = roomNo <= noOfRooms;
    const { adult = 1, children = 0 } = occupancy;

    const onChangeOccupancy = event => dispatch(
        updateRoomConfig(roomNo, { ...occupancy, [event.target.name]: event.target.value })
    );
    const onSelectRoom = event => {
        const checked = event.target.checked;
        if (checked) {
            dispatch(setNoOfRooms(roomNo));
        } else {
            dispatch(setNoOfRooms(roomNo - 1))
        }
    }
    return <RoomCardRoot className={cardClassName}>
        <div className="title">
            <div>
                {roomNo !== 1 && <input type="checkbox" id={checkBoxId} value={roomNo} checked={checked} onChange={onSelectRoom} />}
            </div>
            <label htmlFor={checkBoxId}>Room {roomNo}</label>
        </div>
        <Content>
            <p>Adults (18+)</p>
            <p>Children (0-17)</p>
            <select name="adult" onChange={onChangeOccupancy} value={adult} disabled={disabled}>
                <option value={1}>1</option>
                <option value={2}>2</option>
            </select>
            <select name="children" onChange={onChangeOccupancy} value={children} disabled={disabled}>
                <option value={0}>0</option>
                <option value={1}>1</option>
                <option value={2}>2</option>
            </select>
        </Content>
    </RoomCardRoot>
}

const mapStateToProps = ({ noOfRooms, roomOccupancy }, { roomNo }) => ({
    noOfRooms,
    occupancy: roomOccupancy[roomNo - 1]
})
export default connect(mapStateToProps)(RoomCard);