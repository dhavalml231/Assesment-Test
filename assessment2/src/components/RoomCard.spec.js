import React from "react";
import { create } from "react-test-renderer";
import { RoomCard } from './RoomCard';

describe('Roomcard', () => {
    it('should render first room properly', () => {
        const room = create(<RoomCard roomNo={1} noOfRooms={1} />);
        expect(room.toJSON()).toMatchSnapshot();
    });

    it('should render disabled room properly', () => {
        const room = create(<RoomCard roomNo={3} noOfRooms={1} />);
        expect(room.toJSON()).toMatchSnapshot();
    });

    it('should render non-first enabled room properly', () => {
        const room = create(<RoomCard roomNo={2} noOfRooms={4} />);
        expect(room.toJSON()).toMatchSnapshot();
    });
})