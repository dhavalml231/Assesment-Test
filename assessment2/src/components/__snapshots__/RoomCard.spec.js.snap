// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roomcard should render disabled room properly 1`] = `
<div
  className="disabled sc-bdVaJa ksjZNB"
>
  <div
    className="title"
  >
    <div>
      <input
        checked={false}
        id="3_chkBox"
        onChange={[Function]}
        type="checkbox"
        value={3}
      />
    </div>
    <label
      htmlFor="3_chkBox"
    >
      Room 
      3
    </label>
  </div>
  <div
    className="sc-bwzfXH bFPRny"
  >
    <p>
      Adults (18+)
    </p>
    <p>
      Children (0-17)
    </p>
    <select
      disabled={true}
      name="adult"
      onChange={[Function]}
      value={1}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
    </select>
    <select
      disabled={true}
      name="children"
      onChange={[Function]}
      value={0}
    >
      <option
        value={0}
      >
        0
      </option>
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
    </select>
  </div>
</div>
`;

exports[`Roomcard should render first room properly 1`] = `
<div
  className="sc-bdVaJa ksjZNB"
>
  <div
    className="title"
  >
    <div />
    <label
      htmlFor="1_chkBox"
    >
      Room 
      1
    </label>
  </div>
  <div
    className="sc-bwzfXH bFPRny"
  >
    <p>
      Adults (18+)
    </p>
    <p>
      Children (0-17)
    </p>
    <select
      disabled={false}
      name="adult"
      onChange={[Function]}
      value={1}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
    </select>
    <select
      disabled={false}
      name="children"
      onChange={[Function]}
      value={0}
    >
      <option
        value={0}
      >
        0
      </option>
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
    </select>
  </div>
</div>
`;

exports[`Roomcard should render non-first enabled room properly 1`] = `
<div
  className="sc-bdVaJa ksjZNB"
>
  <div
    className="title"
  >
    <div>
      <input
        checked={true}
        id="2_chkBox"
        onChange={[Function]}
        type="checkbox"
        value={2}
      />
    </div>
    <label
      htmlFor="2_chkBox"
    >
      Room 
      2
    </label>
  </div>
  <div
    className="sc-bwzfXH bFPRny"
  >
    <p>
      Adults (18+)
    </p>
    <p>
      Children (0-17)
    </p>
    <select
      disabled={false}
      name="adult"
      onChange={[Function]}
      value={1}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
    </select>
    <select
      disabled={false}
      name="children"
      onChange={[Function]}
      value={0}
    >
      <option
        value={0}
      >
        0
      </option>
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
    </select>
  </div>
</div>
`;
